lib/lambda_ethereum_consensus/validator/block_builder.ex

```diff
@@ -101,7 +101,16 @@ defmodule LambdaEthereumConsensus.Validator.BlockBuilder do
     # PERF: the state can be cached for the later build_block call
     head_block = Blocks.get_block!(head_root)
     pre_state = BlockStates.get_state!(head_root)
-    head_payload_hash = head_block.body.execution_payload.block_hash
+
+    head_payload_hash =
+      if proposed_slot == 1 do
+        # If the head is the genesis block, fetches the payload_block_hash.
+        with {:ok, %{block_hash: block_hash}} <- ExecutionClient.get_block_metadata(0) do
+          block_hash
+        end
+      else
+        head_block.body.execution_payload.block_hash
+      end
```

Comment:
`with` is not used correctly here, as in an error case `head_payload_hash` will contain an error tuple, which won't be checked anywhere. We should probably do something like this:

```elixir

head_payload_hash =
  if proposed_slot == 1 do
    ExecutionClient.get_block_metadata(0)
  else
    {:ok, block_hash.body.execution_payload}
  end

with {:ok, %{block_hash: block_hash}} <- head_payload_hash
  # The rest of the already existing with
  do
end
```